exercises/bank-account/canonical-data.json

```diff
@@ -0,0 +1,383 @@
+      "property": "bankAccount",
+      "input": {
+        "operations": [
+          {
+            "operation": "open"
+          },
+          {
+            "operation": "open"
+          }
+        ]
+      },
+      "expected": { "error": "account already open" }
+    },
+    {
+      "uuid": "0722d404-6116-4f92-ba3b-da7f88f1669c",
+      "description": "Reopened account does not retain balance",
+      "property": "bankAccount",
+      "input": {
+        "operations": [
+          {
+            "operation": "open"
+          },
+          {
+            "operation": "deposit",
+            "amount": 50
+          },
+          {
+            "operation": "close"
+          },
+          {
+            "operation": "open"
+          },
+          {
+            "operation": "balance"
+          }
+        ]
+      },
+      "expected": 0
+    },
+    {
+      "uuid": "ec42245f-9361-4341-8231-a22e8d19c52f",
+      "description": "Cannot withdraw more than deposited",
+      "property": "bankAccount",
+      "input": {
+        "operations": [
+          {
+            "operation": "open"
+          },
+          {
+            "operation": "deposit",
+            "amount": 25
+          },
+          {
+            "operation": "withdraw",
+            "amount": 50
+          }
+        ]
+      },
+      "expected": { "error": "amount must be less than balance" }
+    },
+    {
+      "uuid": "4f381ef8-10ef-4507-8e1d-0631ecc8ee72",
+      "description": "Cannot withdraw negative",
+      "property": "bankAccount",
+      "input": {
+        "operations": [
+          {
+            "operation": "open"
+          },
+          {
+            "operation": "deposit",
+            "amount": 100
+          },
+          {
+            "operation": "withdraw",
+            "amount": -50
+          }
+        ]
+      },
+      "expected": { "error": "amount must be greater than 0" }
+    },
+    {
+      "uuid": "d45df9ea-1db0-47f3-b18c-d365db49d938",
+      "description": "Cannot deposit negative",
+      "property": "bankAccount",
+      "input": {
+        "operations": [
+          {
+            "operation": "open"
+          },
+          {
+            "operation": "deposit",
+            "amount": -50
+          }
+        ]
+      },
+      "expected": { "error": "amount must be greater than 0" }
+    },
+    {
+      "uuid": "ba0c1e0b-0f00-416f-8097-a7dfc97871ff",
+      "description": "Can handle concurrent transactions",
+      "comments": [
+        "This test is a good candidate to practice concurrency, which",
+        "may not be available or possible in the implementing language track.",
+        "Concurrency testing is often made by running the same operations in",
+        "parallel, and then checking that the final state is as expected."
+      ],
+      "scenarios": ["concurrent"],
+      "property": "bankAccount",
+      "input": {
+        "initial_operations": [
```

Comment:
if the consensus is that this is the best way to do things then I won't stand in your way. It strikes me as:

* inflexible (since now you can only have concurrent operations in between an initial and final section, instead of arbitrary interleavings)
* violating the schema (`property` now may have `input` of different types: it might either have `operations`, or not) 

What I expected to see was:

```json
{
  "input": {
    "operations": [
      { 
        "operation": "open"
      },
      { 
        "operation": "concurrent",
        "number": 1000,
        "operations": [
          { 
            "operation": "deposit",
            "amount": 1
          },
          { 
            "operation": "withdraw",
            "amount": 1
          } 
        ] 
      },
      { 
        "operation": "balance"
      },
    ]
  },
  "expected": 0
}
```

Reply:
I prefer this too. This also allows us to have multiple _batches_ of concurrent operations instead of always _serial start_ _concurrent_ _serial end_.

Reply:
@glennj and @ErikSchierboom, I think we should have a solution we all agree upon so I want to hear your thoughts

Reply:
I'm on board with this suggestion.

Reply:
Me too. That sounds like a better approach